File: LeetCode/1828. Queries on Number of Points Inside a Circle/solution.py
"""
Runtime: 860 ms
Memory Usage: 15.2 MB
"""
class Solution:
    def countPoints(self, points: List[List[int]], queries: List[List[int]]) -> List[int]:
        calc = lambda x, y, r, i, j: 1 if (x - i) * (x - i) + (y - j) * (y - j) <= r * r else 0
        return [sum(calc(x, y, r, i, j) for i, j in points) for x, y, r in queries]

File: LeetCode/1828. Queries on Number of Points Inside a Circle/solution.cpp
/**
 * Runtime: 84 ms
 * Memory Usage: 15.8 MB 
 */
class Solution {
public:
    vector<int> countPoints(vector<vector<int>>& points, vector<vector<int>>& queries) {
        vector<int> ans;
        for (auto &query: queries) {
            int x = query[0], y = query[1], r = query[2];
            int cnt = 0;
            for (auto &point: points) {  
                int i = point[0], j = point[1];
                if ((x - i) * (x - i) + (y - j) * (y - j) <= r * r) {
                    ++ cnt;
                }
            }
            ans.push_back(cnt);
        }
        return ans;        
    }
};
